frontend/app/components/wp-fast-table/state/wp-table-filters.service.ts

```diff
@@ -0,0 +1,128 @@
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+//
+// See doc/COPYRIGHT.rdoc for more details.
+// ++
+
+import {
+  WorkPackageTableBaseService,
+  TableStateStates
+} from './wp-table-base.service';
+import {QueryResource} from '../../api/api-v3/hal-resources/query-resource.service';
+import {QuerySchemaResourceInterface} from '../../api/api-v3/hal-resources/query-schema-resource.service';
+import {QueryFilterResource} from '../../api/api-v3/hal-resources/query-filter-resource.service';
+import {
+  QueryFilterInstanceResource,
+} from '../../api/api-v3/hal-resources/query-filter-instance-resource.service';
+import {CollectionResource} from '../../api/api-v3/hal-resources/collection-resource.service';
+import {opServicesModule} from '../../../angular-modules';
+import {States} from '../../states.service';
+import {State} from '../../../helpers/reactive-fassade';
+import {WorkPackageTableFilters} from '../wp-table-filters';
+
+export class WorkPackageTableFiltersService extends WorkPackageTableBaseService {
+  protected stateName = 'filters' as TableStateStates;
+
+  constructor(public states: States,
+              private $q: ng.IQService) {
+    super(states);
+  }
+
+  public initialize(query:QueryResource, schema:QuerySchemaResourceInterface) {
+    let filters = _.map(query.filters, filter => filter.$copy());
+
+    this.loadCurrentFiltersSchemas(filters).then(() => {
+      let newState = new WorkPackageTableFilters(filters, schema);
+
+      this.state.put(newState);
+    });
+  }
+
+  public get currentState():WorkPackageTableFilters {
+    return this.state.getCurrentValue() as WorkPackageTableFilters;
+  }
+
+  public get current():QueryFilterInstanceResource[]{
+    if (this.currentState) {
+      return this.currentState.current;
+    } else {
+      return [];
+    }
+  }
+
+  public replace(newState:WorkPackageTableFilters) {
+    this.state.put(newState);
+  }
+
+  public remove(removedFilter:QueryFilterInstanceResource) {
+    this.currentState.remove(removedFilter);
+
+    this.state.put(this.currentState);
+  }
+
+  private loadCurrentFiltersSchemas(filters:QueryFilterInstanceResource[]):ng.IPromise<{}> {
+    return this.$q.all(_.map(filters,
+                       (filter:QueryFilterInstanceResource) => this.loadFilterSchema(filter)));
+  }
+
+  private loadFilterSchema(filter:QueryFilterInstanceResource):ng.IPromise<{}> {
+    let deferred = this.$q.defer();
+
+    filter.schema.$load().then(() => {
+      if (filter.values.length &&
+          filter.currentSchema &&
+          filter.currentSchema.values &&
+          filter.currentSchema.values.allowedValues &&
+          (filter.currentSchema.values.allowedValues as CollectionResource)['$load']) {
```

Comment:
Have not tested, but you might have used `_.has(filter, ['values.length', 'currentSchema.values.allowedValues.$load')` instead.